hamocc/mo_carchm.F90

```diff
@@ -0,0 +1,880 @@
+              ocetra(i,j,k,idet14)  = ocetra(i,j,k,idet14) *c14dec
+              ocetra(i,j,k,icalc14) = ocetra(i,j,k,icalc14)*c14dec
+              ocetra(i,j,k,idoc14)  = ocetra(i,j,k,idoc14)*c14dec
+              ocetra(i,j,k,iphy14)  = ocetra(i,j,k,iphy14)*c14dec
+              ocetra(i,j,k,izoo14)  = ocetra(i,j,k,izoo14)*c14dec
+            endif
+
+            ! Save bottom level dissociation konstants for use in sediment module
+            if( k==kbo(i,j) ) then
+
+              keqb( 1,i,j)  = K1
+              keqb( 2,i,j)  = K2
+              keqb( 3,i,j)  = Kb
+              keqb( 4,i,j)  = Kw
+              keqb( 5,i,j)  = Ks1
+              keqb( 6,i,j)  = Kf
+              keqb( 7,i,j)  = Ksi
+              keqb( 8,i,j)  = K1p
+              keqb( 9,i,j)  = K2p
+              keqb(10,i,j)  = K3p
+              keqb(11,i,j)  = Kspc
+
+            endif
+
+          endif ! omask>0.5
+        enddo
+      enddo
+    enddo
+    !$OMP END PARALLEL DO
+
+    ! C14 decay in the sediment (could be moved to sediment part)
+    if (use_cisonew .and. .not. use_sedbypass) then
+      do k=1,ks
+        !$OMP PARALLEL DO PRIVATE(i)
+        do j=1,kpje
+          do i=1,kpie
+            if(omask(i,j).gt.0.5) then
+              sedlay(i,j,k,issso14)=sedlay(i,j,k,issso14)*c14dec
+              sedlay(i,j,k,isssc14)=sedlay(i,j,k,isssc14)*c14dec
+              powtra(i,j,k,ipowc14)=powtra(i,j,k,ipowc14)*c14dec
+            endif
+          enddo
+        enddo
+        !$OMP END PARALLEL DO
+      enddo
+
+      !$OMP PARALLEL DO PRIVATE(i)
+      do j=1,kpje
+        do i=1,kpie
+          if(omask(i,j).gt.0.5) then
+            burial(i,j,issso14) = burial(i,j,issso14)*c14dec
+            burial(i,j,isssc14) = burial(i,j,isssc14)*c14dec
+          endif
+        enddo
+      enddo
+      !$OMP END PARALLEL DO
+    endif ! end of use_cisonew and not use_sedbypass
+
+  end subroutine carchm
+
+  subroutine carchm_kequi(temp,saln,prb,Kh,Khd,K1,K2,Kb,Kw,Ks1,Kf,Ksi,           &
+       K1p,K2p,K3p,Kspc,Kspa)
+
+    !*******************************************************************************
+    ! J. Schwinger,    *BCCR, Bergen*    09.02.16
+    ! Purpose: Calculate equilibrium constant for the carbonate system
+    ! Method:  J.Schwinger,      *Uni Research, Bergen*   2018-04-12
+    ! - added output Khd (CO2 solubility w.r.t. dry air) and Kspa
+    !*******************************************************************************
+
+    use mo_chemcon, only: tzero,rgas,bor1,bor2,salchl,ac1,ac2,ac3,ac4,bc1,bc2,bc3,ad1,ad2,ad3,bd1,bd2,bd3,a0,a1,a2,b0,b1,b2
+
+    ! Arguments
+    real,    intent(in)    :: temp   ! potential temperature [degr C].
+    real,    intent(in)    :: saln   ! salinity [psu].
+    real,    intent(in)    :: prb    ! pressure [bar].
+    real,    intent(out)   :: Kh     ! equilibrium constant Kh  =  [CO2]/pCO2, moist air.
+    real,    intent(out)   :: Khd    ! equilibrium constant Kh  =  [CO2]/pCO2, dry air.
```

Comment:
comment for `Khd` could be fixed: `Kh` -> `Khd` (formerly wrong)

Reply:
Fixed.

Reply:
Sorry being nit-picky, but the comment still shows `Kh` instead of `Khd` - anyway - was our fault initially.